Can't change from a material with vertex color to a material without vertex color.

### Expected behaviour
I change material to any other without issues

### Actual behaviour
If I change the material from one that has a vertex attribute to one without a vertex attribute, I get an error, and my project is broken after that.

### Steps to reproduce
1. Open this project  
[change_material.zip](https://github.com/defold/defold/files/15152630/change_material.zip)
2. open main.collection -> go -> sprite
3. change material to the builtins sprite material

<hr/>

<table>
<tr><td>Build time</td><td>2024-04-29T09:28:08.092096</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>5883d3d4e3f4053fe851fb88b2918ddd19b51e9b</td></tr>
<tr><td>Defold engine sha</td><td>fd3b8c652df601220d8651f581fa2fada8205237</td></tr>
<tr><td>Defold version</td><td>1.8.0</td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/organizations/defold/issues/?query=id%3A"220b06f788ec4bcb8c270603162a9b68"'>220b06f788ec4bcb8c270603162a9b68</a></td></tr>
<tr><td>GPU</td><td>Apple M1 Max</td></tr>
<tr><td>GPU Driver</td><td>2.1 Metal - 88</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>aarch64</td></tr>
<tr><td>OS name</td><td>Mac OS X</td></tr>
<tr><td>OS version</td><td>14.4.1</td></tr>
<table>

```
editor.graphics$resize_doubles.invokeStatic.class java.lang.AssertionError: Assert failed: (keyword? semantic-type)
    at editor.graphics$resize_doubles.invokeStatic(graphics.clj:154)
    at editor.graphics$attribute_value.invokeStatic(graphics.clj:430)
    at editor.graphics$attribute_properties_by_property_key$iter__69312__69316$fn.invoke(graphics.clj:470)
    at clojure.lang.LazySeq.sval(LazySeq.java:42)
    at clojure.lang.LazySeq.seq(LazySeq.java:58)
    at clojure.lang.RT.seq(RT.java:531)
    at clojure.core$seq__5387.invokeStatic(core.clj:137)
    at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24)
    at clojure.core.protocols$fn.invokeStatic(protocols.clj:75)
    at clojure.core.protocols$fn.invoke(protocols.clj:75)
    at clojure.core.protocols$fn$G__8068__8086.invoke(protocols.clj:13)
    at clojure.core$reduce.invokeStatic(core.clj:6828)
    at clojure.core$into.invokeStatic(core.clj:6895)
    at editor.sprite$fn$fn.invoke(sprite.clj:480)
    at clojure.core$update.invokeStatic(core.clj:6196)
    at editor.sprite$fn.invokeStatic(sprite.clj:425)
    at editor.sprite$fn.invoke(sprite.clj:421)
    at clojure.lang.AFn.applyToHelper(AFn.java:154)
    at clojure.lang.AFn.applyTo(AFn.java:144)
    at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
    at clojure.lang.RestFn.invoke(RestFn.java:408)
    at clojure.lang.Var.invoke(Var.java:384)
    at editor.sprite$SpriteNode$behavior$_properties$fn.invoke(sprite.clj:533)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.sprite$SpriteNode$behavior$_properties.invokeStatic(sprite.clj:533)
    at editor.sprite$SpriteNode$behavior$_properties.invoke(sprite.clj:533)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at internal.node$pull_first_input_value.invokeStatic(node.clj:1204)
    at editor.game_object$EmbeddedComponent$behavior$node_outline$fn.invoke(game_object.clj:217)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.game_object$EmbeddedComponent$behavior$node_outline.invokeStatic(game_object.clj:217)
    at editor.game_object$EmbeddedComponent$behavior$node_outline.invoke(game_object.clj:217)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at internal.node$pull_input_values$fn.invoke(node.clj:1220)
    at clojure.core$mapv$fn.invoke(core.clj:6912)
    at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
    at clojure.core$reduce.invokeStatic(core.clj:6827)
    at clojure.core$mapv.invokeStatic(core.clj:6903)
    at internal.node$pull_input_values.invokeStatic(node.clj:1221)
    at editor.game_object$GameObjectNode$behavior$node_outline$fn.invoke(game_object.clj:421)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.game_object$GameObjectNode$behavior$node_outline.invokeStatic(game_object.clj:421)
    at editor.game_object$GameObjectNode$behavior$node_outline.invoke(game_object.clj:421)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at internal.node$pull_first_input_value.invokeStatic(node.clj:1204)
    at editor.collection$EmbeddedGOInstanceNode$behavior$source_outline$fn.invoke(collection.clj:229)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.collection$EmbeddedGOInstanceNode$behavior$source_outline.invokeStatic(collection.clj:229)
    at editor.collection$EmbeddedGOInstanceNode$behavior$source_outline.invoke(collection.clj:229)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at editor.collection$EmbeddedGOInstanceNode$behavior$node_outline$fn.invoke(collection.clj:229)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.collection$EmbeddedGOInstanceNode$behavior$node_outline.invokeStatic(collection.clj:229)
    at editor.collection$EmbeddedGOInstanceNode$behavior$node_outline.invoke(collection.clj:229)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at internal.node$pull_input_values$fn.invoke(node.clj:1220)
    at clojure.core$mapv$fn.invoke(core.clj:6912)
    at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
    at clojure.core$reduce.invokeStatic(core.clj:6827)
    at clojure.core$mapv.invokeStatic(core.clj:6903)
    at internal.node$pull_input_values.invokeStatic(node.clj:1221)
    at editor.collection$CollectionNode$behavior$node_outline$fn.invoke(collection.clj:430)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.collection$CollectionNode$behavior$node_outline.invokeStatic(collection.clj:430)
    at editor.collection$CollectionNode$behavior$node_outline.invoke(collection.clj:430)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at internal.node$pull_first_input_value.invokeStatic(node.clj:1204)
    at editor.app_view$AppView$behavior$active_outline$fn.invoke(app_view.clj:330)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.app_view$AppView$behavior$active_outline.invokeStatic(app_view.clj:330)
    at editor.app_view$AppView$behavior$active_outline.invoke(app_view.clj:330)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at internal.node$pull_first_input_value.invokeStatic(node.clj:1204)
    at editor.outline_view$OutlineView$behavior$root$fn.invoke(outline_view.clj:193)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.outline_view$OutlineView$behavior$root.invokeStatic(outline_view.clj:193)
    at editor.outline_view$OutlineView$behavior$root.invoke(outline_view.clj:193)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at editor.outline_view$OutlineView$behavior$tree_view$fn.invoke(outline_view.clj:193)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.outline_view$OutlineView$behavior$tree_view.invokeStatic(outline_view.clj:193)
    at editor.outline_view$OutlineView$behavior$tree_view.invoke(outline_view.clj:193)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at editor.outline_view$OutlineView$behavior$tree_selection_root_its$fn.invoke(outline_view.clj:193)
    at internal.node$trace_expr.invokeStatic(node.clj:44)
    at editor.outline_view$OutlineView$behavior$tree_selection_root_its.invokeStatic(outline_view.clj:193)
    at editor.outline_view$OutlineView$behavior$tree_selection_root_its.invoke(outline_view.clj:193)
    at clojure.lang.Var.invoke(Var.java:393)
    at internal.node.NodeImpl.produce_value(node.clj:295)
    at internal.node$node_value.invokeStatic(node.clj:454)
    at internal.system$node_value.invokeStatic(system.clj:354)
    at dynamo.graph$do_node_value.invokeStatic(graph.clj:830)
    at dynamo.graph$node_value.invokeStatic(graph.clj:833)
    at editor.outline_view$root_iterators.invokeStatic(outline_view.clj:284)
    at editor.outline_view$fn$fnk108976_positional__108981.invoke(outline_view.clj:356)
    at editor.outline_view$fn$fnk108976__108993$fn.invoke(outline_view.clj:356)
    at editor.outline_view$fn$fnk108976__108993.invoke(outline_view.clj:356)
    at clojure.lang.AFn.applyToHelper(AFn.java:154)
    at clojure.lang.AFn.applyTo(AFn.java:144)
    at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
    at clojure.lang.RestFn.invoke(RestFn.java:408)
    at editor.handler$invoke_fnk.invokeStatic(handler.clj:256)
    at editor.handler$enabled_QMARK_.invokeStatic(handler.clj:296)
    at editor.ui$refresh_menu_item_state.invokeStatic(ui.clj:1681)
    at editor.ui$refresh_menu_item_state.invoke(ui.clj:1657)
    at editor.ui$refresh_menu_item_state.invokeStatic(ui.clj:1663)
    at editor.ui$refresh_menubar_state.invokeStatic(ui.clj:1694)
    at editor.ui$refresh_menus_BANG_.invokeStatic(ui.clj:1844)
    at editor.ui$refresh.invokeStatic(ui.clj:1865)
    at editor.app_view$refresh_menus_and_toolbars_BANG_.invokeStatic(app_view.clj:1801)
    at editor.app_view$make_app_view$fn.invoke(app_view.clj:1925)
    at editor.ui$__GT_timer$fn$fn.invoke(ui.clj:1968)
    at clojure.lang.AFn.run(AFn.java:22)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
    at java.security.AccessController.doPrivileged(AccessController.java:399)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
    at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
```

--- a/editor/src/clj/editor/graphics.clj
+++ b/editor/src/clj/editor/graphics.clj
@@ -153,7 +153,6 @@
 
 (defn resize-doubles [double-values semantic-type ^long new-element-count]
   {:pre [(vector? double-values)
-         (keyword? semantic-type)
          (nat-int? new-element-count)]}
   (let [old-element-count (count double-values)]
     (cond

--- a/editor/src/clj/editor/graphics.clj
+++ b/editor/src/clj/editor/graphics.clj
@@ -153,6 +153,7 @@
 
 (defn resize-doubles [double-values semantic-type ^long new-element-count]
   {:pre [(vector? double-values)
+         (or (nil? semantic-type) (keyword? semantic-type))
          (nat-int? new-element-count)]}
   (let [old-element-count (count double-values)]
     (cond

